payment-failure/src/main/scala/com/gu/identity/paymentfailure/Lambda.scala

```diff
@@ -1,10 +1,57 @@
 package com.gu.identity.paymentfailure
 
 import com.typesafe.scalalogging.StrictLogging
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import scala.collection.JavaConverters._
 
 object Lambda extends StrictLogging {
 
-  def handler(): Unit = {
-    logger.info("executing lambda")
+  def getConfig: Either[Throwable, Config] = {
+    val optionConfig = for {
+      idapiHost <- Option(System.getenv("idapiHost"))
+      brazeApiHost <- Option(System.getenv("brazeApiHost"))
+      idapiAccessToken <- Option(System.getenv("idapiAccessToken"))
+      sqsQueueUrl <- Option(System.getenv("sqsQueueUrl"))
+    } yield {
+      Config(idapiHost, brazeApiHost, idapiAccessToken, sqsQueueUrl)
+    }
+
+    optionConfig match {
+      case Some(config) => Right(config)
+      case _ => Left(new Exception(s"Missing or incorrect config. Please check environment variables"))
+    }
+  }
+
+  def handler(event: SQSEvent, context: Context): Unit = {
+
+    logger.info(s"context :  $context")
+    logger.info(s"event :  $event")
+    logger.info(s"received message batch of size: ${event.getRecords.size}")
+
+    process(event).foreach {
+      case Left(throwable) => throw throwable
+      case _ => logger.info("process successful")
+    }
+  }
+
+  def process(event: SQSEvent): List[Either[Throwable, BrazeResponse]]= {
+    val messages = event.getRecords.asScala.toList
+
+    messages.map( mes => {
+      for {
+        config <- getConfig
+
+        identityClient = new IdentityClient(config)
+        sqsService = new SqsService(config)
+        brazeClient = new BrazeClient
+        sendEmailService = new SendEmailService(identityClient, brazeClient)
```

Comment:
@Mullefa is there where you would pass the config in? Doesn't seem right to me...

Reply:
You could create the services at the top level within the lambda:
```scala
object Lambda {
  val config = getConfig.valueOr(throw _)
  val identityClient = new IdentityClient(config)
  // etc etc
}
```
[Documentation](https://typelevel.org/cats/api/cats/syntax/EitherOps.html#valueOr[BB%3E:B](f:A=%3EBB):BB) for `valueOr()`.

Reply:
```scala
  val config =  new EitherOps[Throwable, Config](getConfig).valueOr(throw _)
```
?

Reply:
Yeah exactly - but if you `import cats.syntax.either._` the either instance will be converted implicitly to an `EitherOps` instance.